locator/load_sketch.hh

```diff
@@ -70,10 +70,13 @@ public:
         : _tm(std::move(tm)) {
     }
 
-    future<> populate(std::optional<host_id> host = std::nullopt) {
+    future<> populate(std::optional<host_id> host = std::nullopt, std::optional<table_id> only_table = std::nullopt) {
         const topology& topo = _tm->get_topology();
         co_await utils::clear_gently(_nodes);
         for (auto&& [table, tmap_] : _tm->tablets().all_tables()) {
+            if (only_table && *only_table != table) {
```

Comment:
Why do it in the inner loop?
If only_table is provided, we can look it up directly in `all_tables()`, no?
If so, I'd refactor the inner loop out into a function and call it once if `only_table` is provided, or for each table otherwise.

Reply:
> Why do it in the inner loop?

For simplicity of the change, used only in tests so efficiency doesn't matter. But I reworked it.